### PR TITLE
polarity: init at unstable-2024-06-28

### PR DESCRIPTION
**Pull request by MangoIV**
## Description of changes

This adds the polarity research language to nixpkgs

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/po/polarity/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchFromGitHub {
+    owner = "polarity-lang";
+    repo = "polarity";
+    rev = "59bd7a2c3c3d0a61b25d3bb85b9d21f7b3fef343";
+    hash = "sha256-85uo2GAGxWAWwN2vyhUqwz28Ofb+2eOSuetzovAle+A=";
+  };
+
+  cargoLock = {
+    lockFile = "${src}/Cargo.lock";
```
Comment (NobbZ) [inline pkgs/by-name/po/polarity/package.nix:18]: This is IFD, please vendor the lock file.

File: pkgs/by-name/po/polarity/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "polarity";
+  version = "unstable-2024-06-28";
```
Comment (NobbZ) [inline pkgs/by-name/po/polarity/package.nix:8]: `0-unstable-2024-06-28` unless there happened actual releases.

**Review by NobbZ — CHANGES_REQUESTED** [review]

File: pkgs/by-name/po/polarity/package.nix (hunk 1 above)
Comment (MangoIV) [reply pkgs/by-name/po/polarity/package.nix:18]: alright :P

**Review by NobbZ — APPROVED** [review]

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 323373` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>polarity</li>
  </ul>
</details>

**Pull request merged**